backend/lucene/src/main/java/org/hibernate/search/backend/lucene/types/sort/comparatorsource/impl/LuceneTextFieldComparatorSource.java

```diff
@@ -32,7 +32,7 @@ public LuceneTextFieldComparatorSource(String nestedDocumentPath, Object missing
 
 	@Override
 	public FieldComparator<?> newComparator(String fieldname, int numHits, int sortPos, boolean reversed) {
-		final boolean sortMissingLast = missingLast() ^ reversed;
+		final boolean sortMissingLast = missingLast() ^ reversed && !missingLowest() || missingHighest();
```

Comment:
I don't know about you, but I don't read this and immediately know what's the precedence of `^` vs `&&`... 

Anyway, this code was already unnecessarily complex before, maybe it's time for a change.

First, we should rename the variable to `considerMissingHighest`, because that's what it actually means; see how its value is interpreted in `org.apache.lucene.search.FieldComparator.TermOrdValComparator#TermOrdValComparator(int, java.lang.String, boolean)`.

Second, maybe these if/else blocks would be clearer?

```suggestion
		final boolean considerMissingHighest;
		if ( SortMissingValue.MISSING_LOWEST.equals( missingValue ) ) {
			considerMissingHighest = false;
		}
		else if ( SortMissingValue.MISSING_HIGHEST.equals( missingValue ) ) {
			considerMissingHighest = true;
		}
		else if ( SortMissingValue.MISSING_LAST.equals( missingValue ) ) {
			// To appear last, missing values must be considered highest, or lowest if the order is reversed.
			considerMissingHighest = !reversed;
		}
		else { // SortMissingValue.MISSING_FIRST, the default
			// To appear first, missing values must be considered lowest, or highest if the order is reversed.
			considerMissingHighest = reversed;
		}
```